### DISCUSSION TITLE
Better inline await pattern

### DISCUSSION THREAD

**Discussion by ctigrisht** [dotnet/csharplang#7474] [Language Ideas]
I have encountered the need to do the following quite a bit: `var username = (await service.GetUserAsync()).Username;`

I would love to be able to do an inline await such as: `var username = service.GetUserAsync():await.Username`.

This would be very useful for saving a few lines of code. Note that I used the `:` symbol to not break compatibility with old programs where the object might contain a `.await` field or property.

**Comment by CyrusNajmabadi** [comment]
You don't need the `:`.  In an `async` method, `await` is *always* a keyword (even after a `.`).

**Reply by ctigrisht** [reply]
It is, but wouldn't the compiler assume you are awaiting what comes after the keyword, and not before? It would also bring errors if `await` is a field or property of what is being awaited

**Reply by CyrusNajmabadi** [reply]
The compiler can have whatever rules it wants here.

**Reply by HaloFour** [reply]
The compiler reserves `await` as a keyword anywhere in an `async` method, so in order to even reference a member with the name `await` requires prefixing with `@`.

**Comment by HaloFour** [comment]
See: #4076

**Reply by ctigrisht** [reply]
This was closed, and this proposal is different, I'm proposing a `:` instead of `.`

**Reply by HaloFour** [reply]
Minor syntax differences doesn't change the nature of the proposal.

I'll note that `:` will very likely interact poorly with the ternary conditional operator.

**Reply by CyrusNajmabadi** [reply]
> Minor syntax differences doesn't change the nature of the proposal.

Agreed.  The crux of why we did not go this direction is that we didn't feel that `await` was the postfix operator we wanted at all.  And that we always wanted to see how this played out with DUs.

**Reply by ctigrisht** [reply]
@CyrusNajmabadi what would be the operator then?
